src/AutorestSwift/View Models/ServiceClientViewModel.swift

```diff
@@ -42,23 +46,184 @@ struct ParameterViewModel {
+            self.knownMediaType = nil
+        }
+
+        // check if the request body is from signature parameter. If yes, store the object type
+        // and add the request signature parameter to operation parameters
+        var params = [ParameterViewModel]()
+        for param in request.signatureParameters ?? [] {
+            params.append(ParameterViewModel(from: param))
+        }
+        self.params = params
+
+        if request.signatureParameters?.count == 1,
+            request.signatureParameters?[0].schema.type == AllSchemaTypes.object {
+            self.objectType = request.signatureParameters?[0].schema.name
+        } else {
+            self.objectType = nil
+        }
+    }
+}
+
+struct ResponseViewModel {
+    let statusCodes: [String]
+    let knownMediaType: String?
+    let mediaTypes: [String]?
+    let objectType: String?
+
+    init(from response: Response) {
+        if let httpResponse = (response.protocol.http as? HttpResponse) {
+            var statusCodes = [String]()
+            for code in httpResponse.statusCodes {
+                statusCodes.append(code.rawValue)
+            }
+            self.statusCodes = statusCodes
+            self.knownMediaType = httpResponse.knownMediaType?.rawValue
+            self.mediaTypes = httpResponse.mediaTypes
+        } else {
+            self.statusCodes = []
+            self.knownMediaType = nil
+            self.mediaTypes = nil
+        }
+
+        // check if the request body schema type is object, store the object type of the response body
+        if let schemaResponse = response as? SchemaResponse,
+            schemaResponse.schema.type == AllSchemaTypes.object {
+            self.objectType = schemaResponse.schema.name
+        } else {
+            self.objectType = nil
+        }
+    }
+}
+
 struct OperationViewModel {
     let name: String
     let comment: ViewModelComment
     let params: [ParameterViewModel]
     let returnType: ReturnTypeViewModel?
+    let queryParams: [KeyValueViewModel]?
+    let headerParams: [KeyValueViewModel]?
+    let uriParams: [KeyValueViewModel]?
+    let requests: [RequestViewModel]?
+    let responses: [ResponseViewModel]?
+    let method: String?
+    let path: String?
 
     init(from schema: Operation) {
         self.name = operationName(for: schema.name)
         self.comment = ViewModelComment(from: schema.description)
 
+        // operation parameters should be all the signature paramters from the schema
+        // and the signature parameters of the Request
         var items = [ParameterViewModel]()
         for param in schema.signatureParameters ?? [] {
             items.append(ParameterViewModel(from: param))
         }
+
+        var queryParams = [KeyValueViewModel]()
+        var headerParams = [KeyValueViewModel]()
+        var uriParams = [KeyValueViewModel]()
+
+        for param in schema.parameters ?? [] {
+            if let httpParam = (param.protocol.http as? HttpParameter?) {
```

Comment:
```suggestion
            if let httpParam = (param.protocol.http as? HttpParameter) {
```
Generally the point of the `if let foo = bar as? Bar` pattern is to have the unwrapped value in the body.